activerecord/lib/active_record/attribute_set.rb

```diff
@@ -64,9 +64,7 @@ def freeze
     end
 
     def deep_dup
-      dup.tap do |copy|
-        copy.instance_variable_set(:@attributes, attributes.deep_dup)
-      end
+      self.class.new(attributes.deep_dup)
```

Comment:
This is avoiding the wasted shallow dup in `initialize_dup` below, right?

Does `dup.tap do |copy| copy.attributes.transform_values!(&:deep_dup) end` bench similarly?

My only hesitation about using `new` is that while it *currently* saves an allocation, it seems more likely to become the costlier of the two at some point in the future.